src/hooks/carousel/__tests__/useCarousel.browser.spec.js

```diff
@@ -0,0 +1,119 @@
+import '&test/browser/styles/main.css';
+import React, { useRef, useState } from 'react';
+import { Box } from '&test/browser/utils/Box';
+import { mountHookWithUI } from '&test/browser/utils/mountHookWithUI';
+import { useCarousel } from '../useCarousel';
+
+const mainStyle = `
+  .list {
+    width: 700px;
+    max-width: 100%;
+  }
+`;
+
+const useWrappedHook = (result) => {
+  const ref = useRef();
+  const [selected, setSelected] = useState(0);
+  result.current = { selected, setSelected };
+
+  const carousel = useCarousel(ref, { selected, onSelect: setSelected, numTiles: 4 }, []);
+
+  return (
+    <>
+      <div>
+        <button type="button" {...carousel.props.arrow.left}>left</button>
+        <button type="button" {...carousel.props.arrow.right}>right</button>
+      </div>
+      <div className="list" ref={ref} {...carousel.props.list}>
+        <Box id="box-1" {...carousel.props.tile(0)}><button type="button">yolo</button></Box>
+        <Box id="box-2" {...carousel.props.tile(1)}><button type="button">yolo</button></Box>
+        <Box id="box-3" {...carousel.props.tile(2)}><button type="button">yolo</button></Box>
+        <Box id="box-4" {...carousel.props.tile(3)}><button type="button">yolo</button></Box>
+        <div className="margin-fix" data-lwc-ignore />
+      </div>
+      <p>
+        {carousel.pages.current} / {carousel.pages.total}
+      </p>
+    </>
+  );
+};
+
+const mountTest = (props) =>
+  mountHookWithUI(useWrappedHook, props, mainStyle);
+
+it('selects tile on click or with arrow keys', () => {
+  mountTest().its('result').as('result');
```

Comment:
This is interesting syntax. So is this because you many need updated from whatever provides the results? Not just a reference to what it is now? 🤔

Reply:
So this is adding the alias `result` for the `result` field that is returned from `mountTest`, so that we can get that `result` field later & do assertions against it. If we don't alias it here, then there is no way to retrieve the field later in the test (just saving it to a variable doesn't work because cypress commands are run asynchronously with some magic behind the scenes - [docs](https://docs.cypress.io/guides/core-concepts/introduction-to-cypress.html#Commands-Are-Asynchronous))

https://docs.cypress.io/guides/core-concepts/variables-and-aliases.html#Aliases